hostfactory/1.1/providerplugins/azurecc/scripts/src/cyclecloud_provider.py

```diff
@@ -759,6 +770,27 @@ def _create_status(self, input_json, json_writer=None):
                 if request_id not in requests_store:
                     logger.warning("Unknown request_id %s. Creating a new entry and resetting requestTime", request_id)
                     requests_store[request_id] = {"requestTime": calendar.timegm(self.clock())}
+                #set default
+                actual_machine_cnt = len(all_nodes)
+                if not requests_store[request_id].get("lastNumNodes") :
+                    requests_store[request_id]["lastNumNodes"] = actual_machine_cnt
+                if actual_machine_cnt < requests_store[request_id]["lastNumNodes"]:
+                    logger.debug("Actual machine count %s is less than previous machines %s", actual_machine_cnt, requests_store[request_id]["lastNumNodes"])
+                    request_envelope = self.capacity_tracker.get_request(request_id)
+                    if not request_envelope:
+                        logger.warning("Out-of-capacity detected")
+                        logger.warning("No request envelope found for request id %s", request_id)
+                    else:
+                        reqs = request_envelope.get('sets', [])
+                        if not reqs:
+                            continue
```

Comment:
This conditions seems worthy of a log statement. It is basically equivalent to the "if not request_envelope", no?